### PR TITLE
Auth landing page

### PR DESCRIPTION
**Pull request by jennifer-lu**
## Notion ticket link

[\[Auth\] Implement user sign on to Landing Page](https://www.notion.so/uwblueprintexecs/Auth-Implement-user-sign-on-to-Landing-Page-88ac6787cc224ff487340c23528ce8b7)

## Implementation description
* Use `Auth.currentAuthenticatedUser()` to get authenticated user information
* Display user's name if authenticated

## Steps to test
1. Go to http://localhost:3000/
2. Observe that header says "WELCOME TO YOUR TOOLKIT"
3. Go to http://localhost:3000/login
4. Log in, or sign up and log in (note the name you use)
5. Observe redirect to http://localhost:3000/
6. Observe that header says "WELCOME TO YOUR TOOLKIT, \[NAME\]"
7. Please also take a look at [documentation](https://www.notion.so/uwblueprintexecs/Authentication-455747d7459f4e969c768837621aa5bb) (see User section)

## What should reviewers focus on?
* Does it work?
* Does implementation make sense?

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Comment by jennifer-lu** [comment]
I added the `amplify` directory to `eslintignore` because linting was failing. Please let me know if this is an ok solution or if I should use a different fix.

**Comment by ChinemeremChigbo** [comment]
> I added the `amplify` directory to `eslintignore` because linting was failing. Please let me know if this is an ok solution or if I should use a different fix.

For me running yarn lint --fix worked

- I removed eslint from .eslint ignore
image

- changed the single quotes in package.json to escaped double quotes as per ([why single quotes to escaped double quotes?](https://stackoverflow.com/questions/54543063/how-can-i-suppress-the-no-files-matching-the-pattern-message-in-eslint))
image

and then ran yarn lint --fix from the project's root directory

**Comment by jennifer-lu** [comment]
> For me running yarn lint --fix worked
>
> ...
>
> and then ran yarn lint --fix from the project's root directory
>
> and everything still seems to run as usual:

@ChinemeremChigbo  Awesome, I'll make those changes. Thanks so much for the solution!

**Review by ChinemeremChigbo — APPROVED** [review]
Sweet! Looks great, thanks for making those changes.

**Review by chamod-gamage — CHANGES_REQUESTED** [review]
Great work!
Not sure if this is in scope but the password requirement string has a typo (should be `lowercase`) and also can we make it wrap such that it doesn't increase the width of the input above it (the UX is kind of jarring when it makes the input super wide).
image
There's also [some issues with the demographics form](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=6295f60944244182923b0160acd3cee6) but I believe that's out of scope for this PR.

**Comment by jennifer-lu** [comment]
> Great work! Not sure if this is in scope but the password requirement string has a typo (should be `lowercase`) and also can we make it wrap such that it doesn't increase the width of the input above it (the UX is kind of jarring when it makes the input super wide).

Thanks for the review! I agree that the login UI should be polished, but I don't think it's in the scope of this ticket.

File: pages/index.tsx [hunk 1]
```diff
@@ -192,10 +193,22 @@ const Tools: React.FC<{ variant: string }> = ({ variant }) => {
...
 const Home: React.FC = () => {
     const [showAlert, setShowAlert] = useState(true);
+    const [user, setUser] = useState<any | null>(null);
+
     const variant = useMediaQuery({ query: `(max-width: 925px)` })
         ? "mobile"
         : "desktop";
 
+    useEffect(() => {
+        Auth.currentAuthenticatedUser()
```
Comment (tawnyzhao) [inline pages/index.tsx:203]: You can wrap the asynchronous code inside a use effect in an async function to use async await syntax

**Review by tawnyzhao — APPROVED** [review]

**Comment by jennifer-lu** [comment]
@tawnyzhao I've fixed the gitignore and removed the amplify files. Also changed the `useEffect` code. Please double check if you have time.

@chamod-gamage I also think that the login UI can be polished, but it isn't in the scope of this ticket. The fixes that you mentioned can be made in another pr (would appreciate it if you unrequest changes so I can merge).

**Review by tawnyzhao — APPROVED** [review]

**Review by chamod-gamage — APPROVED** [review]
Awesome work!

**Pull request merged**